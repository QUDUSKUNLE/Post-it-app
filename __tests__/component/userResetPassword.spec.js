import React from 'react';
import { shallow } from 'enzyme';
import ResetPassword
  from '../../client/src/js/components/userResetPassword.jsx';

describe('PostIt-app', () => {
  it('expects App component to be defined', () => {
    expect(ResetPassword).toBeDefined();
  });
  it('should render correctly', () => {
    const component = shallow(<ResetPassword/>);
    expect(component).toMatchSnapshot();
  });
});
