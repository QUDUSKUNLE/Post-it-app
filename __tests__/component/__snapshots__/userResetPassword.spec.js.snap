// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostIt-app should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <nav
      className="navbar navbar-inverse navabar-fixed-top"
      role="navigation"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <button
            className="navbar-toggle collapsed"
            data-target=".navbar-collapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
          <Link
            className="navbar-brand"
            replace={false}
            to="#"
          >
            PostIt
            <small>
              App
            </small>
          </Link>
        </div>
        <div
          className="collapse navbar-collapse"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <Link
                replace={false}
                to="/signin"
              >
                Sign in
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      className="passwordreset"
    >
      <center>
        <h4>
          Reset your password
        </h4>
      </center>
      <br />
      <div
        className="container resetform"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 col-md-offset-3 w3-card w3-white"
          >
            <form
              id="resetform"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Enter your email address
                </label>
                <input
                  className="signinform"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="johndoe@example.com"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <button
                className="signinformbtn"
                type="submit"
              >
                Send
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <nav
        className="navbar navbar-inverse navabar-fixed-top"
        role="navigation"
      >
        <div
          className="container"
        >
          <div
            className="navbar-header"
          >
            <button
              className="navbar-toggle collapsed"
              data-target=".navbar-collapse"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="sr-only"
              >
                Toggle navigation
              </span>
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
            </button>
            <Link
              className="navbar-brand"
              replace={false}
              to="#"
            >
              PostIt
              <small>
                App
              </small>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign in
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        className="passwordreset"
      >
        <center>
          <h4>
            Reset your password
          </h4>
        </center>
        <br />
        <div
          className="container resetform"
        >
          <div
            className="row"
          >
            <div
              className="col-md-6 col-md-offset-3 w3-card w3-white"
            >
              <form
                id="resetform"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Enter your email address
                  </label>
                  <input
                    className="signinform"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="johndoe@example.com"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <button
                  className="signinformbtn"
                  type="submit"
                >
                  Send
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ResetPassword />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ResetPassword {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "email": "",
          "response": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div
            className="passwordreset"
          >
            <center>
              <h4>
                Reset your password
              </h4>
            </center>
            <br />
            <div
              className="container resetform"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6 col-md-offset-3 w3-card w3-white"
                >
                  <form
                    id="resetform"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Enter your email address
                      </label>
                      <input
                        className="signinform"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="johndoe@example.com"
                        required={true}
                        type="email"
                        value=""
                      />
                    </div>
                    <button
                      className="signinformbtn"
                      type="submit"
                    >
                      Send
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div
            className="passwordreset"
          >
            <center>
              <h4>
                Reset your password
              </h4>
            </center>
            <br />
            <div
              className="container resetform"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-6 col-md-offset-3 w3-card w3-white"
                >
                  <form
                    id="resetform"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Enter your email address
                      </label>
                      <input
                        className="signinform"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="johndoe@example.com"
                        required={true}
                        type="email"
                        value=""
                      />
                    </div>
                    <button
                      className="signinformbtn"
                      type="submit"
                    >
                      Send
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ResetPassword />,
}
`;
